ControlSettings/ControlSettingsTV.cpp

```diff
@@ -0,0 +1,3332 @@
+        if(pqmode.empty())
+            pqmode = "current";
+
+        source = parameters.HasLabel("source") ? parameters["source"].String() : "";
+        if(source.empty())
+            source = "current";
+
+        format = parameters.HasLabel("format") ? parameters["format"].String() : "";
+        if(format.empty())
+            format = "current";
+			
+        GetParamIndex(source,pqmode,format,sourceIndex,pqIndex,formatIndex);
+        err = GetLocalparam("Backlight",formatIndex,pqIndex,sourceIndex,backlight);
+        
+        if( err == 0 ) {
+            response["backlight"] = std::to_string(backlight);
+            LOGINFO("Exit : Backlight Value: %d \n", backlight);
+            returnResponse(true, "success");
+        }
+        else {
+            returnResponse(false, getErrorString(tvERROR_GENERAL).c_str());
+        }
+    }
+
+    uint32_t ControlSettingsTV::getBacklight2(const JsonObject& parameters, JsonObject& response)
+    {
+        LOGINFO("Entry");
+
+        std::string pqmode;
+        std::string source;
+        std::string format;
+        std::string key;
+        JsonObject range;
+        JsonObject backlightObj;
+        range["From"] = 0;
+        range["To"] = 100;
+        backlightObj["Range"] = range;
+        int sourceIndex=0,pqIndex=0,formatIndex=0;
+        tvError_t ret = tvERROR_NONE;
+        int backlight = 0,err = 0;
+
+        pqmode = parameters.HasLabel("pictureMode") ? parameters["pictureMode"].String() : "";
+        if(pqmode.empty())
+            pqmode = "current";
+
+        source = parameters.HasLabel("source") ? parameters["source"].String() : "";
+        if(source.empty())
+            source = "current";
+
+        format = parameters.HasLabel("format") ? parameters["format"].String() : "";
+        if(format.empty())
+            format = "current";
+
+        GetParamIndex(source,pqmode,format,sourceIndex,pqIndex,formatIndex);
+        err = GetLocalparam("Backlight",formatIndex,pqIndex,sourceIndex,backlight);
+        if( err == 0 ) {
+            backlightObj["Setting"] = std::to_string(backlight);
+            response["Backlight"] = backlightObj;
+            LOGINFO("Exit : Backlight Value: %d \n", backlight);
+            returnResponse(true, "success");
+        }
+        else {
+            returnResponse(false, getErrorString(tvERROR_GENERAL).c_str());
+        }
+    }
+
+    uint32_t ControlSettingsTV::setBacklight(const JsonObject& parameters, JsonObject& response)
+    {
+        LOGINFO("Entry\n");
+     
+	std::string value;
+        std::string pqmode;
+        std::string source;
+        std::string format;
+        int backlight = 0;
+        int userScaleBacklight = 0;
+        tvError_t ret  = tvERROR_NONE;
+
+        value = parameters.HasLabel("backlight") ? parameters["backlight"].String() : "";
+        returnIfParamNotFound(value);
+        backlight = std::stoi(value);
```

Comment:
what's the point in passing input parameters as text and converting them to number if you can pass number?